camelot/view/controls/delegates/customdelegate.py

```diff
@@ -147,14 +148,19 @@ def get_standard_item(cls, locale, model_context):
 
         :param locale: the `QLocale` to be used to display locale dependent values
         :param model_context: a FieldActionModelContext object
-        
+        :param with_action_states: When True, the ActionStatesRole data will be set.
+                                   This can be disabled if a delegate handles it's own action states.
+
         :return: a `QStandardItem` object
         """
         routes = model_context.field_attributes.get('action_routes', [])
         states = []
         for action in model_context.field_attributes.get('actions', []):
-            state = action.get_state(model_context)
-            states.append(dataclasses.asdict(state))
+            if isinstance(action, FieldAction):
+                state = action.get_state(model_context)
+                states.append(dataclasses.asdict(state))
+            else:
```

Comment:
in principe zouden we hier dan niet mogen komen ?  dwz dat die isinstance eerder een assert mag zijn ?